### PR TITLE
feat(kms): public key conversion to JWK format.

### PR DESCRIPTION
**Pull request by mdetemad**
Add public key conversion to JWK format.

### CODE REVIEW

File: kms/get_public_key_jwk.go [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+	block, _ := pem.Decode(key)
+	_, err = x509.ParsePKIXPublicKey(block.Bytes)
+	if err != nil {
+		return fmt.Errorf("failed to parse public key: %v", err)
+	}
+
+	// If all above checks pass, convert it into JWK format.
+	jwkKey, err := jwk.ParseKey(key, jwk.WithPEM(true))
+	if err != nil {
+		fmt.Printf("Failed to parse the PEM public key: %s\n", err)
```
Comment (telpirion) [inline kms/get_public_key_jwk.go:81]: issue: same as above--return raw error or use `%w`.

File: kms/get_public_key_jwk.go [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+		return nil
+	}
+
+	json.NewEncoder(w).Encode(jwkKey)
```
Comment (telpirion) [inline kms/get_public_key_jwk.go:85]: nit: consider printing out some meaningful message to the `io.Writer`. That way customers can see that the code "worked" when they copy/paste/run it in their own IDE.

File: kms/get_public_key_jwk.go [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+	"hash/crc32"
+	"io"
+
+	kms "cloud.google.com/go/kms/apiv1"
+	"cloud.google.com/go/kms/apiv1/kmspb"
+	"github.com/lestrrat-go/jwx/v2/jwk"
+)
+
+// getPublicKeyJwk retrieves the public key from an asymmetric key pair on Cloud KMS.
+func getPublicKeyJwk(w io.Writer, name string) error {
```
[the remark this answers is not in the document]
Comment (mdetemad) [reply kms/get_public_key_jwk.go:33]: I could update this but all the existing modules use `name`. 
What do you think?

File: kms/get_public_key_jwk.go [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+	// name := "projects/my-project/locations/us-east1/keyRings/my-key-ring/cryptoKeys/my-key/cryptoKeyVersions/123"
+
+	// Create the client.
+	ctx := context.Background()
+	client, err := kms.NewKeyManagementClient(ctx)
+	if err != nil {
+		return fmt.Errorf("failed to create kms client: %v", err)
```
[the remark this answers is not in the document]
Comment (mdetemad) [reply kms/get_public_key_jwk.go:40]: Again all the existing modules return errors of this form.

File: kms/get_public_key_jwk.go [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+
+	// Build the request.
+	req := &kmspb.GetPublicKeyRequest{
+		Name: name,
+	}
+
+	// Call the API to get the public key.
+	result, err := client.GetPublicKey(ctx, req)
+	if err != nil {
+		return fmt.Errorf("failed to get public key: %v", err)
```
[the remark this answers is not in the document]
Comment (mdetemad) [reply kms/get_public_key_jwk.go:52]: Again all the existing modules (e.g., get_public_key.go) return errors of this form.

File: kms/get_public_key_jwk.go [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+	if int64(crc32c(key)) != result.PemCrc32C.Value {
+		return fmt.Errorf("getPublicKey: response corrupted in-transit")
+	}
+
+	// Optional - parse the public key.
+	// This transforms the string key into a Go PublicKey.
+	block, _ := pem.Decode(key)
+	_, err = x509.ParsePKIXPublicKey(block.Bytes)
+	if err != nil {
+		return fmt.Errorf("failed to parse public key: %v", err)
```
[the remark this answers is not in the document]
Comment (mdetemad) [reply kms/get_public_key_jwk.go:75]: Again all the existing modules (e.g., get_public_key.go) return errors of this form.

File: kms/get_public_key_jwk.go (hunk 1 above)
Comment (mdetemad) [reply kms/get_public_key_jwk.go:81]: Again all the existing modules (e.g., get_public_key.go) return errors of this form.

File: kms/get_public_key_jwk.go (hunk 2 above)
Comment (mdetemad) [reply kms/get_public_key_jwk.go:85]: Actually we write the result into `w` which is an `io.Writer` object.
Do you want something explicit?

File: kms/get_public_key_jwk.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (telpirion) [reply kms/get_public_key_jwk.go:33]: Please provide a more descriptive name. A bad habit committed in the past is no reason to commit the same bad habit now.

File: kms/get_public_key_jwk.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (telpirion) [reply kms/get_public_key_jwk.go:40]: Please fix anyway.

File: kms/get_public_key_jwk.go (hunk 6 above)
[the remark this answers is not in the document]
Comment (telpirion) [reply kms/get_public_key_jwk.go:75]: Please fix anyway.

File: kms/get_public_key_jwk.go (hunk 1 above)
Comment (telpirion) [reply kms/get_public_key_jwk.go:81]: Please fix anyway.

File: kms/get_public_key_jwk.go (hunk 2 above)
Comment (telpirion) [reply kms/get_public_key_jwk.go:85]: Yes. For example:

```go
var buf bytes.Buffer
json.NewEncoder(&buf).Encode(jwkKey)

newKey := buf.String()
fmt.Fprintf(w, "New key created:\n%s", newKey)
```

**Review by telpirion — CHANGES_REQUESTED** [review]
Please address comments. Thank you!

File: kms/get_public_key_jwk.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (mdetemad) [reply kms/get_public_key_jwk.go:40]: could you help me understand this better?

File: kms/get_public_key_jwk.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (mdetemad) [reply kms/get_public_key_jwk.go]: Done, thanks for the suggestions.

File: kms/get_public_key_jwk.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (mdetemad) [reply kms/get_public_key_jwk.go]: Done.

File: kms/get_public_key_jwk.go (hunk 5 above)
[the remark this answers is not in the document]
Comment (mdetemad) [reply kms/get_public_key_jwk.go]: Done.

File: kms/get_public_key_jwk.go (hunk 6 above)
[the remark this answers is not in the document]
Comment (mdetemad) [reply kms/get_public_key_jwk.go]: Done.

File: kms/get_public_key_jwk.go (hunk 1 above)
Comment (mdetemad) [reply kms/get_public_key_jwk.go]: Done.

File: kms/get_public_key_jwk.go (hunk 2 above)
Comment (mdetemad) [reply kms/get_public_key_jwk.go:85]: Done, thanks for the suggestions.

**Review by telpirion — APPROVED** [review]

**Comment by telpirion** [comment]
Looks like there's a linter issue--please fix and we'll merge this PR. Thank you!

**Comment by mdetemad** [comment]
> Looks like there's a linter issue--please fix and we'll merge this PR. Thank you!

Are we good now?

**Pull request merged**